src/tigerbeetle/inspect.zig

```diff
@@ -0,0 +1,1100 @@
+//! Decode a TigerBeetle data file without running a replica or modifying the data file.
+//! This tool adheres to the "be liberal in what you accept" side of Postel's Law.
+
+const std = @import("std");
+const assert = std.debug.assert;
+const log = std.log.scoped(.inspect);
+
+const cli = @import("cli.zig");
+const vsr = @import("vsr");
+const stdx = vsr.stdx;
+const fatal = vsr.flags.fatal;
+const schema = vsr.lsm.schema;
+const constants = vsr.constants;
+const tb = vsr.tigerbeetle;
+const Storage = vsr.storage.Storage;
+const SuperBlockHeader = vsr.superblock.SuperBlockHeader;
+const SuperBlock = vsr.SuperBlockType(Storage);
+const StateMachine = vsr.state_machine.StateMachineType(Storage, constants.state_machine_config);
+const Grid = vsr.GridType(Storage);
+const BlockPtr = vsr.grid.BlockPtr;
+const BlockPtrConst = vsr.grid.BlockPtrConst;
+const allocate_block = vsr.grid.allocate_block;
+const is_composite_key = vsr.lsm.composite_key.is_composite_key;
+
+pub fn main(allocator: std.mem.Allocator, cli_args: *const cli.Command.Inspect) !void {
+    var stdout_buffer = std.io.bufferedWriter(std.io.getStdOut().writer());
+
+    const stdout = stdout_buffer.writer();
+
+    var inspector = try Inspector.init(allocator, cli_args.path);
+    defer inspector.deinit();
+
+    switch (cli_args.query) {
+        .superblock => try inspector.inspect_superblock(stdout),
+        .wal => |args| {
+            if (args.slot) |slot| {
+                if (slot >= constants.journal_slot_count) {
+                    return fatal(
+                        "--slot: slot exceeds {}",
+                        .{constants.journal_slot_count - 1},
+                    );
+                }
+                try inspector.inspect_wal_slot(stdout, slot);
+            } else {
+                try inspector.inspect_wal(stdout);
+            }
+        },
+        .replies => |args| {
+            if (args.slot) |slot| {
+                if (slot >= constants.clients_max) {
+                    return fatal("--slot: slot exceeds {}", .{constants.clients_max - 1});
+                }
+                try inspector.inspect_replies_slot(stdout, args.superblock_copy, slot);
+            } else {
+                try inspector.inspect_replies(stdout, args.superblock_copy);
+            }
+        },
+        .grid => |args| {
+            if (args.superblock_copy >= constants.superblock_copies) {
+                return fatal(
+                    "--superblock-copy: copy exceeds {}\n",
```

Comment:
We _shouldn't_ need a `\n` here?

Reply:
Oops; those were converted from `writeAll` and I forgot to remove the newline.